src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -0,0 +1,170 @@
+  import scala.collection.JavaConversions.asScalaIterator
+
+  Io.assertReadable(input)
+  output.foreach(out => Io.assertListable(out.getParent))
+
+  def execute(): Unit = {
+    val progress = new ProgressLogger(logger, verb="written", unit=1e6.toInt)
+    val in       = SamReaderFactory.make.open(input.toFile)
+    val iterator = in.iterator()
+    val counter  = new ErrorRateByReadPositionCounter
+
+    iterator.filterNot { rec =>
+      rec.getReadUnmappedFlag || rec.isSecondaryOrSupplementary ||
+        rec.getDuplicateReadFlag || rec.getReadFailsVendorQualityCheckFlag
+    }
+    .foreach { rec =>
+      counter.accept(rec)
+      progress.record(rec)
+    }
+
+    val path = PathUtil.pathTo(output.getOrElse(PathUtil.removeExtension(input)) + ErrorRateByReadPositionMetric.FileExtension)
+    Metric.write[ErrorRateByReadPositionMetric](metrics=counter.metrics, path=path)
+
+    in.safelyClose()
+  }
+}
+
+object ErrorRateByReadPositionMetric {
+  val FileExtension = ".error_rate_by_read_position.txt"
+}
+
+/** Gives the error rate for read one, read two, and across both reads, for the given position relative to the start of
+  * the read. For paired end reads, the read position is relative to the first base sequenced for the given end of the
+  * read.
+  *
+  * @param position the read position (1-based).
+  * @param error_rate_read_one the error rate of the first read.
+  * @param error_rate_read_two the error rate of the second read.
+  * @param error_rate_total the error rate across all reads at this read position.
+  * @param bases_read_one the total number of bases observed at this position for read one.
+  * @param bases_read_two the total number of bases observed at this position for read two.
+  * @param bases_total the total number of bases observed at this position across all reads.
+  */
+case class ErrorRateByReadPositionMetric
+( position: Int,
+  error_rate_read_one: Double,
+  error_rate_read_two: Double,
+  error_rate_total: Double,
+  bases_read_one: Long,
+  bases_read_two: Long,
+  bases_total: Long
+) extends Metric
+
+private class ErrorRateByReadPositionCounter {
+  val r1TotalBasesByReadPosition = new NumericCounter[Int]()
+  val r1ErrorBasesByReadPosition = new NumericCounter[Int]()
+  val r2TotalBasesByReadPosition = new NumericCounter[Int]()
+  val r2ErrorBasesByReadPosition = new NumericCounter[Int]()
+
+  def accept(rec: SAMRecord): Unit = {
+    val first: Boolean    = !rec.getReadPairedFlag || rec.getFirstOfPairFlag
+    val totalBasesByReadPosition = if (first) r1TotalBasesByReadPosition else r2TotalBasesByReadPosition
+    val errorBasesByReadPosition = if (first) r1ErrorBasesByReadPosition else r2ErrorBasesByReadPosition
+    val read              = maybeReverseComplement(rec.getReadBases.clone(), rc=rec.getReadNegativeStrandFlag)
+    val reference         = maybeReverseComplement(SequenceUtil.makeReferenceFromAlignment(rec, false), rc=rec.getReadNegativeStrandFlag)
```

Comment:
This depends on the MD tag right?  I would strongly prefer to just use the reference sequence directly than rely on the MD tag being present and correct.